crates/wasmtime/src/runtime/vm/instance.rs

```diff
@@ -1120,13 +1120,13 @@ impl Instance {
                     TableInitialValue::Null { precomputed } => precomputed,
                     TableInitialValue::Expr(_) => unreachable!(),
                 };
-                let func_index = precomputed.get(i as usize).cloned();
+                let func_index = precomputed.get(usize::try_from(i).unwrap()).cloned();
```

Comment:
Also to clarify this `try_from(...).unwrap()` is ok (and the one below). Panicking here helps catch bugs rather than silently truncating by accident. Thanks for updating this!